server/src/main/java/org/opensearch/identity/Principals.java

```diff
@@ -20,11 +21,42 @@ public enum Principals {
     UNAUTHENTICATED(new StringPrincipal("Unauthenticated"));
 
     private final Principal principal;
+    private final SecretKey secretKey;
+    private final byte[] initializationVector; 
+    public static final String ALGORITHM = "AES";
+    public static final int KEY_SIZE = 128;
+    public static final int INITIALIZATION_VECTOR_SIZE = 96;
+    public static final int TAG_LENGTH = 128;
 
     private Principals(final Principal principal) {
         this.principal = principal;
+        this.initializationVector = generateIV(); 
+        try{ 
+            this.secretKey = generateSecretKey();
```

Comment:
Won't this generate keys for every instance of this class? Even if this is a singleton class, wouldn't this generate a different key for every node of the cluster? 

We may need to look into how [`SecureSetting`](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/SecureSetting.java) stores secrets and think about storage and configuration for these keys.

Reply:
It will generate a new key and initialization vector for every principal. This is the only way I am aware of to make sure that each principal is provided a different encoding. Otherwise, the encodings would only be different for each extension but would remain the same for all the different principals on a given extension. 

I take it this is not what we want? 

For the creation of the instances on each node, I am not sure how to handle this as currently I do not know what type of data sharing between nodes we have--I can look at the SecureSettings class but we do not currently have any principal lookup or storage system that I am aware of.

Reply:
Ignore this. I am not sure why the PR continues to show this file change when it no longer exists in the merging branch but these changes are gone. Sorry for the confusion.